### PR TITLE
chore: upgrade h2 database

### PR DESCRIPTION
**Pull request by kasiaMarek**
resolves: https://github.com/scalameta/metals/issues/5447

Do we want to fallback to deleting the db if this mechanism fails?

### CODE REVIEW

**Comment by tgodzik** [comment]
> resolves: #5447
> 
> Do we want to fallback to deleting the db if this mechanism fails?

That seems to be a good ide :+1:

File: metals/src/main/scala/scala/meta/internal/metals/Tables.scala [hunk 1]
```diff
@@ -161,6 +168,47 @@ final class Tables(
...
   }
 
+  /**
+   * Between h2 "2.1.x" and "2.2.x" write/read formats in MVStore changed
+   * (https://github.com/h2database/h2database/pull/3834)
+   */
+  private def upgradeIfNeeded(url: String): Unit = {
+    val oldVersion = 214
+    val formatVersionChangedMessage =
+      "The write format 2 is smaller than the supported format 3"
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/Tables.scala:178]: Any chance we can match on an exact error?

**Review by tgodzik — APPROVED** [review]
Great work! Let's merge this after the next release!

File: metals/src/main/scala/scala/meta/internal/metals/Tables.scala (hunk 1 above)
Comment (kasiaMarek) [reply metals/src/main/scala/scala/meta/internal/metals/Tables.scala:178]: We could not check the exact message and just the error code. Trouble is that we wouldn't know the versions  are`2` and `3`, so if it changes again in the future, it might be `3` and `4`. Maybe it doesn't matter so much since it will fail if the versions change again anyway. Otherwise (looking at the source code), the versions in the error are only in the message I'm afraid.

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala [hunk 2]
```diff
@@ -908,11 +908,10 @@ class MetalsLspService(
   val isInitialized = new AtomicBoolean(false)
 
   def initialized(): Future[Unit] = {
-    loadFingerPrints()
     registerNiceToHaveFilePatterns()
-    tables.connect()
```
Comment (jkciesluk) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala:913]: Why is this line deleted?

Comment (kasiaMarek) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala:913]: `loadFingerPrints()` has to do `tables.connect()` anyway. It's just a duplication really.
